### PR TITLE
[MRESOLVER-327] Allow use of system properties to configure HttpClient

### PR DESCRIPTION
**Pull request by cstamas**
But do not encourage users about this, quite the opposite, drive them toward documentation how to configure transports, for example this page: https://maven.apache.org/guides/mini/guide-proxies.html

---

https://issues.apache.org/jira/browse/MRESOLVER-327

### CODE REVIEW

File: src/site/markdown/configuration.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Option | Type | Description | Default Value | Supports Repo ID Suffix
 `aether.connector.http.preemptivePutAuth` | boolean | Should HTTP client use preemptive-authentication for HTTP PUTs only (works only w/ BASIC). By default is enabled (same as Wagon). | `true` | yes
 `aether.connector.http.retryHandler.count` | int | The maximum number of times a request to a remote HTTP server should be retried in case of an error. | `3` | yes
 `aether.connector.http.supportWebDav` | boolean | If enabled, transport makes best effort to deploy to WebDAV server. This mode is not recommended, better use real Maven Repository Manager instead. | `false` | yes
+`aether.connector.http.useSystemProperties` | boolean | If enabled, underlying Apache HttpClient will use system properties as well to configure itself (typically used to set up HTTP Proxy via Java system properties). This mode is not recommended, better use proper Maven configuration means instead. | `false` | yes
```
[the remark this answers is not in the document]
Comment (kwin) [reply src/site/markdown/configuration.md:44]: I wouldn’t expect the term Maven configuration in here. Rather resolver configuration as resolver can still be used outside Maven

[the remark this answers is not in the document]
Comment (cstamas) [reply src/site/markdown/configuration.md:44]: Originally I wanted to WARN if `proxy != null` (set few lines above), but this builder option is NOT ONLY about proxy but many many other things. Hence the hint "typically used..." as I cannot nor I want to commit to any guarantees here. Simply put: if this option used, user should use it only for "typically set" stuff, and ensure the code is doing what he things is doing.

The reason of existence of this PR is ONLY to allow use of transport-http for use case like https://issues.apache.org/jira/browse/MNG-7721 where user defines only 5 http proxy related properties.

[the remark this answers is not in the document]
Comment (cstamas) [reply src/site/markdown/configuration.md:44]: I consider having both set an error (and at one point I had code even fail), but later realized to simple "delegate" all the responsibility to use: "user knows what he does"

[the remark this answers is not in the document]
Comment (michael-o) [reply src/site/markdown/configuration.md:44]: The description must link to the constructor of HttpClientBuilder which will describe which properties from the system will be used.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/site/markdown/configuration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cstamas) [reply src/site/markdown/configuration.md]: Done

**Comment by cstamas** [comment]
Am merging this as doco was updated, and is also safe as all this is disabled by default. The only reason for this change is to make possible exact use case like the reported in https://issues.apache.org/jira/browse/MNG-7721 to work with transport-http, nothing more and nothing less.

**Pull request merged**